packages/design-system/src/components/Button/Button.js

```diff
@@ -26,20 +26,19 @@ const Button = React.forwardRef((props, ref) => {
     ariaExpanded,
     className,
     type: htmlType,
+    text,
+    iconPosition,
     ...buttonProps
   } = props;
 
   const isDisabled = disabled || loading;
   const type = submit ? 'submit' : htmlType || 'button';
-  let buttonType = null;
-
-  if (primary) {
-    buttonType = 'primary';
-  } else if (destructive) {
-    buttonType = 'destructive';
-  } else if (secondary) {
-    buttonType = 'secondary';
-  }
+  const buttonType =
+    (primary && 'primary') ||
+    (destructive && 'destructive') ||
+    (secondary && 'secondary') ||
+    (text && 'text') ||
```

Comment:
`text` property name might be not the best here. It is usually used to pass some content to the component.

Reply:
carbon has interesting name for this type of a button: "ghost" :D https://www.carbondesignsystem.com/components/button/usage

Reply:
Maybe we should migrate buttons to `type="text"` etc. (but keep current types declaration with boolean props as deprecated way). WDYT?

Reply:
I'd rather die than make changes again in AA ;x 
If we can support current props and the new `type` prop in the same time it could be a good idea. I saw that there is a few cases in AA where it's not handy to have `primary`, `secondary` etc props. The `type` property might be better.

Reply:
Yeah, I had the same issues. I think we should use `type` or `kind` prop and keep `primary` etc. as deprecated, but still usable ofc.

Reply:
Done in 99b10e2c47c91f12884be7178dde5774cedf65b5